indico/modules/rb/client/js/common/bookings/BookingExportModal.jsx

```diff
@@ -0,0 +1,130 @@
+  const errors = {};
+  if (!format) {
+    errors.type = Translate.string('Please choose a file format to export the bookings.');
+  }
+  if (!dates || !Object.values(dates).every(x => x)) {
+    errors.dates = Translate.string('Please choose a valid period.');
+  }
+  if (!rooms || !rooms.length) {
+    errors.rooms = Translate.string('Please choose at least one room to export its bookings.');
+  }
+  return errors;
+}
+
+const BookingExportModal = props => {
+  const {
+    rooms,
+    onClose,
+    actions: {exportBookings},
+  } = props;
+
+  const handleSubmit = async formData => {
+    const rv = await exportBookings(formData);
+    if (rv.error) {
+      return rv.error;
+    }
+    location.href = rv.data.url;
+  };
+
+  const renderModalContent = fprops => {
+    const {submitting, hasValidationErrors, pristine, submitSucceeded} = fprops;
+    return (
+      <>
+        <Modal.Header>
+          <Translate>Export bookings</Translate>
+        </Modal.Header>
+        <Modal.Content>
+          <Form id="export-bookings-form" onSubmit={fprops.handleSubmit} success={submitSucceeded}>
+            <h5>
+              <Translate>File format</Translate>
+            </h5>
+            <Form.Group>
+              <FinalRadio name="format" value="csv" label={Translate.string('CSV')} />
+              <FinalRadio name="format" value="xlsx" label={Translate.string('XLSX')} />
+            </Form.Group>
+            <FinalDatePeriod name="dates" label={Translate.string('Period')} required allowNull />
+            <FinalRoomSelector name="rooms" label={Translate.string('Rooms to export')} required />
+          </Form>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button
+            type="submit"
+            form="export-bookings-form"
+            disabled={submitting || hasValidationErrors || pristine || submitSucceeded}
+            loading={submitting}
+            primary
+          >
+            <Translate>Export</Translate>
+          </Button>
+          <Button type="button" onClick={onClose}>
+            <Translate>Close</Translate>
+          </Button>
+        </Modal.Actions>
+      </>
+    );
+  };
+
+  return (
+    <Modal open onClose={onClose} size="tiny" closeIcon>
+      <FinalForm
+        validate={validate}
+        onSubmit={handleSubmit}
+        render={renderModalContent}
+        initialValues={{rooms, dates: null}}
+        initialValuesEqual={_.isEqual}
+        subscription={{
+          submitting: true,
+          hasValidationErrors: true,
+          pristine: true,
+          submitSucceeded: true,
+        }}
+      />
+    </Modal>
+  );
+};
+
+BookingExportModal.propTypes = {
+  rooms: PropTypes.array,
+  onClose: PropTypes.func.isRequired,
+  actions: PropTypes.exact({
+    exportBookings: PropTypes.func.isRequired,
+  }).isRequired,
+};
+
+BookingExportModal.defaultProps = {
+  rooms: [],
+};
+
+export default connect(
+  null,
+  dispatch => ({
+    actions: bindActionCreators(
```

Comment:
For simple components that don't have tons of actions (let's say no more than 2 or 3?) I probably wouldn't nest them inside `actions`. Because that way you can get this much simpler version:

```js
export default connect(
  null,
  {exportBookings: bookingActions.exportBookings}
)(BookingExportModal);
```

Reply:
However, I don't see a reason to use Redux here at all. There is no benefit in not sending the request directly from within the component.

If you move, the request into the component you can use the `handleSubmitError` util to convert errors to the format final-form expects:

```js
    try {
      response = await indicoAxios.XXX(....);
    } catch (error) {
      return handleSubmitError(error);
    }
```